sql/core/src/main/scala/org/apache/spark/sql/execution/UnionLoopExec.scala

```diff
@@ -157,9 +191,6 @@ case class UnionLoopExec(
           cause = null)
       }
 
-      // Inherit stats and constraints from the dataset of the previous iteration.
-      val prevPlan = LogicalRDD.fromDataset(prevDF.queryExecution.toRdd, prevDF, prevDF.isStreaming)
-        .newInstance()
       unionChildren += prevPlan
 
       currentNumRows += prevCount.toInt
```

Comment:
not related to this PR, but `.toInt` has the risk of overflow. Shall we make `currentNumRows` a long as well?